static/main.js

```diff
@@ -13,32 +21,38 @@ function init() {
     input.addEventListener("input", reprocess, false);
     permalink.addEventListener("click", setSearch);
 
-    let match = (/(?:^\?|&)text=([^&]*)/).exec(window.location.search);
-    if (match) {
-        input.value = decodeURIComponent(match[1]);
+    let match;
+    if ((match = argument("text"))) {
+        input.value = decodeURIComponent(match[2]);
     }
 
     const macros = {};
-    // TODO: Add toggle for displayMode.
-    // https://github.com/Khan/KaTeX/issues/1035
     const options = {displayMode: true};
-    const macroRegex = /(?:^\?|&)(?:\\|%5[Cc])([A-Za-z]+)=([^&]*)/g;
+    const macroRegex = argumentRegex("(?:\\\\|%5[Cc])[A-Za-z]+|.");
     let macroString = "";
     while ((match = macroRegex.exec(window.location.search)) !== null) {
         options.macros = macros;
-        macros["\\" + match[1]] = decodeURIComponent(match[2]);
+        macros[match[1].replace(/%5[cC]/, "\\")] = decodeURIComponent(match[2]);
         macroString += "&" + match[0].substr(1);
     }
 
+    // Use `display=0` or `displayMode=0` to turn off displayMode.
+    if ((match = argument("display|displayMode"))) {
+        if (match[2] === "false" || match[2] === "0" || match[2] === "no") {
```

Comment:
I think `n` and maybe `f` are more common.

Reply:
I think these are fine.  TBH I would've been okay with only checking for `displayMode=0`.

Reply:
I added support for anything starting with 0, n, or f.  Hopefully less guesswork in the future.